mpd_client/src/commands/definitions.rs

```diff
@@ -938,6 +938,189 @@ enum TagTypesAction {
+    }
+}
+
+impl Command for StickerSet {
+    type Response = res::Empty;
+
+    fn into_command(self) -> RawCommand {
+        RawCommand::new("sticker")
+            .argument("set")
+            .argument("song")
+            .argument(self.uri)
+            .argument(self.name)
+            .argument(self.value)
+    }
+}
+
+/// `sticker delete` command
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct StickerDelete {
+    uri: String,
+    name: String,
+}
+
+impl StickerDelete {
+    /// Delete the sticker `name` for the song at `uri`
+    pub fn new(uri: String, name: String) -> Self {
+        Self { uri, name }
+    }
+}
+
+impl Command for StickerDelete {
+    type Response = res::Empty;
+
+    fn into_command(self) -> RawCommand {
+        RawCommand::new("sticker")
+            .argument("delete")
+            .argument("song")
+            .argument(self.uri)
+            .argument(self.name)
+    }
+}
+
+/// `sticker list` command
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct StickerList {
+    uri: String,
+}
+
+impl StickerList {
+    /// Lists all stickers on the song at `uri`
+    pub fn new(uri: String) -> Self {
+        Self { uri }
+    }
+}
+
+impl Command for StickerList {
+    type Response = res::StickerList;
+
+    fn into_command(self) -> RawCommand {
+        RawCommand::new("sticker")
+            .argument("list")
+            .argument("song")
+            .argument(self.uri)
+    }
+}
+
+/// Operator for full (filtered) version
+/// of `sticker find` command
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum StickerFindOperator {
+    /// = operator
+    Equals,
+    /// < operator
+    LessThan,
+    /// > operator
+    GreaterThan,
+}
+
+/// `sticker find` command
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct StickerFind {
+    uri: String,
+    name: String,
+    operator: Option<StickerFindOperator>,
+    value: Option<String>,
+}
+
+impl StickerFind {
+    /// Lists all stickers on the song at `uri`
+    pub fn new(uri: String, name: String) -> Self {
+        Self {
+            uri,
+            name,
+            operator: None,
+            value: None,
+        }
+    }
+
+    /// Find stickers where their value is equal to `value`
+    pub fn where_eq(&self, value: String) -> Self {
+        self.r#where(StickerFindOperator::Equals, value)
+    }
+
+    /// Find stickers where their value is greater than `value`
+    pub fn where_gt(&self, value: String) -> Self {
+        self.r#where(StickerFindOperator::GreaterThan, value)
+    }
+
+    /// Find stickers where their value is less than `value`
+    pub fn where_lt(&self, value: String) -> Self {
+        self.r#where(StickerFindOperator::LessThan, value)
+    }
+
+    fn r#where(&self, operator: StickerFindOperator, value: String) -> Self {
+        Self {
+            name: self.name.clone(),
+            uri: self.uri.clone(),
+            operator: Some(operator),
+            value: Some(value),
+        }
+    }
```

Comment:
These methods should take the command as `self` rather than borrowing. This currently results in needlessly cloning the parameters for the common case of constructing a command and only using it once.

I would also prefer the internal method to use a name that does not require raw identifier syntax, just to not be unnecessarily confusing to readers of the code. Maybe something like `operator`?

Reply:
Opted for `add_filter` as I think that makes it clearer there's an operator/value pair.